### PR TITLE
Package overhaul/clean up of go-talib

### PR DESCRIPTION
**Pull request by xtda**
# PR Description

This is a first round pass at a general clean up of the go-talib package for use within GCT

This splits different indicators into seperate files (under the same package) to make them easier to maintain and group 

Also enableds go module support and github actions for CI testing

## Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [x] go test ./... -race
- [x] golangci-lint run

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Travis with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by xtda** [comment]
noticed some inconsistencies with data sets after i expended to include larger / real data have flagged this as reconstructing while i do further testing today

File: indicators/talib_test.go [hunk 1]
```diff
@@ -0,0 +1,771 @@
...
+
+	// testCrossunder1 = []float64{1, 2, 3, 4, 8, 6, 7}
+	// testCrossunder2 = []float64{1, 1, 10, 9, 5, 3, 7}
+
+	// testNothingCrossed1 = []float64{1, 2, 3, 4, 8, 6, 7}
+	// testNothingCrossed2 = []float64{1, 4, 5, 9, 5, 3, 7}
+	// testCrossover1 = []float64{1, 3, 2, 4, 8, 6, 7}
+	// testCrossover2 = []float64{1, 5, 1, 4, 5, 6, 7}
```
Comment (shazbert) [inline indicators/talib_test.go]: Why are these commented?

File: indicators/talib_test.go [hunk 2]
```diff
@@ -0,0 +1,771 @@
...
+// 			t.Errorf("Closes error: Expected %f at cell %d got %f ", expected, i, resultClose[i])
+// 		}
+// 	}
+//
+// 	for i, expected := range expectedLows {
+// 		if resultLow[i] != expected {
+// 			t.Errorf("Lows error: Expected %f at cell %d got %f ", expected, i, resultLow[i])
+// 		}
+// 	}
+// }
```
Comment (shazbert) [inline indicators/talib_test.go]: reinstate?

File: indicators/talib_test.go [hunk 3]
```diff
@@ -0,0 +1,771 @@
...
+// 	}
+
+// 	// Crossover
+// 	series1 = []float64{1, 3, 2, 4, 8, 6, 7}
+// 	series2 = []float64{1, 5, 1, 4, 5, 6, 7}
+
+// 	if Crossunder(series1, series2) == true {
+// 		t.Error("Crossunder: Not expected and found")
+// 	}
+// }
```
Comment (shazbert) [inline indicators/talib_test.go]: reinstate?

**Review by shazbert — CHANGES_REQUESTED** [review]
Awesome cleanup will test soon.

File: indicators/talib_test.go (hunk 3 above)
Comment (xtda) [reply indicators/talib_test.go]: :+1: forgot to update with new data will do that now

File: indicators/talib_test.go (hunk 2 above)
Comment (xtda) [reply indicators/talib_test.go]: :+1: forgot to update with new data will do that now

**Comment by xtda** [comment]
Looks like Obv values are still off (Thanks @gloriousCode for testing !)  i have gone ahead and added reconstructing while i investigate again but feel free to review other indicators if you wish

**Review by shazbert — COMMENTED** [review]
Looks good, just can't build locally just yet, will do that soon.

**Comment by VarunBatraIT** [comment]
Grouping doesn't support adding up the volume - did you notice?

**Comment by xtda** [comment]
n/a moved to github.com/thrasher-corp/gct-ta

**Pull request closed without merging**